File: Module.Persist.TPM/Model/History/HistoricalPromoSales.cs
using Core.History;
using Module.Persist.TPM.Model.TPM;
using System;

namespace Module.Persist.TPM.Model.History
{
    [AssociatedWith(typeof(PromoSales))]
    public class HistoricalPromoSales : BaseHistoricalEntity<System.Guid>
    {
        public Guid? ClientId { get; set; }
        public Guid? BrandId { get; set; }
        public Guid? BrandTechId { get; set; }
        public Guid? PromoStatusId { get; set; }
        public Guid? MechanicId { get; set; }
        public Guid? BudgetItemId { get; set; }
        public DateTimeOffset? StartDate { get; set; }
        public DateTimeOffset? EndDate { get; set; }
        public DateTimeOffset? DispatchesStart { get; set; }
        public DateTimeOffset? DispatchesEnd { get; set; }
        public int? Number { get; set; }
        public string Name { get; set; }
        public string ClientCommercialSubnetCommercialNetName { get; set; }
        public string BrandName { get; set; }
        public string BrandTechName { get; set; }
        public string PromoStatusName { get; set; }
        public string MechanicMechanicName { get; set; }
        public int? MechanicDiscount { get; set; }
        public string MechanicComment { get; set; }
        public int? Plan { get; set; }
        public int? Fact { get; set; }
        public string BudgetItemBudgetName { get; set; }
        public string BudgetItemName { get; set; }
    }
}

File: Module.Frontend.TPM/Client/model/tpm/promosales/HistoricalPromoSales.js
Ext.define('App.model.tpm.promosales.HistoricalPromoSales', {
    extend: 'Ext.data.Model',
    idProperty: '_Id',
    breezeEntityType: 'HistoricalPromoSales',
    fields: [
        { name: '_Id', type: 'string', hidden: true },
        { name: '_ObjectId', hidden: true },
        { name: '_User', type: 'string', isDefault: true },
        { name: '_Role', type: 'string', isDefault: true },
        { name: '_EditDate', type: 'date', isDefault: true, timeZone: +3, convert: dateConvertTimeZone},
        { name: '_Operation', type: 'string', isDefault: true },
        { name: 'ClientId', useNull: true, hidden: true, isDefault: true },
        { name: 'BrandId', useNull: true, hidden: true, isDefault: true },
        { name: 'BrandTechId', useNull: true, hidden: true, isDefault: true },
        { name: 'PromoStatusId', useNull: true, hidden: true, isDefault: true },
        { name: 'MechanicId', useNull: true, hidden: true, isDefault: true },
        { name: 'BudgetItemId', useNull: true, hidden: true, isDefault: true },
        { name: 'Number', type: 'int', hidden: false, isDefault: true, isKey: true },
        { name: 'Name', type: 'string', hidden: false, isDefault: true },
        { name: 'StartDate', useNull: true, type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone},
        { name: 'EndDate', useNull: true, type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'DispatchesStart', useNull: true, type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone},
        { name: 'DispatchesEnd', useNull: true, type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'ClientCommercialSubnetCommercialNetName', type: 'string' },
        { name: 'BrandName', type: 'string' },
        { name: 'BrandTechName', type: 'string' },
        { name: 'PromoStatusName', type: 'string' },
        { name: 'PromoStatusSystemName', type: 'string' },
        { name: 'MechanicName', type: 'string' },
        { name: 'MechanicDiscount', type: 'int' },
        { name: 'MechanicTypeName', type: 'string' },
        { name: 'Plan', useNull: true, type: 'int', hidden: false, isDefault: true },
        { name: 'Fact', useNull: true, type: 'int', hidden: false, isDefault: true },
        { name: 'BudgetItemBudgetName', type: 'string' },
        { name: 'BudgetItemName', type: 'string' }
    ],
    proxy: {
        type: 'breeze',
        resourceName: 'HistoricalPromoSaleses',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});
